api/src/middleware/authenticate.ts

```diff
@@ -45,7 +50,18 @@ export const handler = async (req: Request, _res: Response, next: NextFunction)
 		return next();
 	}
 
-	req.accountability = await getAccountabilityForToken(req.token, defaultAccountability);
+	try {
+		req.accountability = await getAccountabilityForToken(req.token, defaultAccountability);
+	} catch (err) {
+		if (isDirectusError(err, ErrorCode.InvalidCredentials)) {
```

Comment:
We should also clear the cookie on `InvalidToken` error

```suggestion
		if (isDirectusError(err, ErrorCode.InvalidCredentials) || isDirectusError(err, ErrorCode.InvalidToken)) {
```

Reply:
Afaik the invalid token error can only be thrown if the user tampered with the session cookie, or in the insanely small time window between cookie expiration in the browser and session token expectation in the backend. And I think it is fair to have a failed state here.

Reply:
No, if the `SECRET` changed for some reason, the invalid token error is also thrown. This should not lock out users.